File: app/src/main/java/com/imangazalievm/bubbble/domain/interactors/ShotZoomInteractor.java
package com.imangazalievm.bubbble.domain.interactors;


import com.imangazalievm.bubbble.domain.repository.ImagesRepository;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ShotZoomInteractor {

    private ImagesRepository imagesRepository;

    @Inject
    public ShotZoomInteractor(ImagesRepository imagesRepository) {
        this.imagesRepository = imagesRepository;
    }

    public Completable saveImage(String shotImageUrl) {
        return imagesRepository.saveImage(shotImageUrl);
    }

}

File: app-mvp/app/src/main/java/com/bubbble/domain/shotzoom/ShotZoomInteractor.kt
package com.bubbble.domain.shotzoom

import com.bubbble.data.images.ImagesRepository
import javax.inject.Inject

class ShotZoomInteractor @Inject constructor(
    private val imagesRepository: ImagesRepository
) {

    suspend fun saveImage(shotImageUrl: String) {
        imagesRepository.saveImage(shotImageUrl)
    }

}